src/engine/macros.rs

```diff
@@ -23,10 +23,14 @@ macro_rules! destroy_filesystems {
     ( $s:ident; $fs:expr ) => {
         let mut removed = Vec::new();
         for uuid in $fs.iter().map(|x| *x) {
+            match $s.filesystems.get_mut_by_uuid(&uuid) {
```

Comment:
Ok

The root was that teardown() had an un-needed "&mut self".  Thanks for the help.